pmacApp/src/pmacController.cpp

```diff
@@ -4150,9 +4165,109 @@ asynStatus pmacController::readDeviceType() {
+    status = asynError;
+    debug(DEBUG_ERROR, functionName, "Error reading card cpu");
+  }
+
+  if (status == asynSuccess) {
+    if (strcmp(cpu_.c_str(),"PowerPC,460EX") == 0) {
+      cpuNumCores_ = 1;
+    } else if (strcmp(cpu_.c_str(),"PowerPC,APM86xxx") == 0) {
+      cpuNumCores_ = 2;
+    } else if (strcmp(cpu_.c_str(),"arm,LS1021A") == 0) {
+      strcpy(cmd, "Sys.Cores");
+      status = pBroker_->immediateWriteRead(cmd, reply);
+      status = parseIntegerVariable(cmd, reply,
+                                    "Read the number of CPU cores", cpuNumCores_);
+    } else {
+      debug(DEBUG_ERROR, functionName, "CPU not recognized");
+    }
+  }
+
+  if (status == asynSuccess) {
+    status = setIntegerParam(PMAC_C_CpuNumCores_, cpuNumCores_);
+  }
+
   return status;
 }
 
+asynStatus pmacController::getTasksCore() {
+  asynStatus status = asynSuccess;
+  char reply[PMAC_MAXBUF];
+  char cmd[PMAC_MAXBUF];
+  static const char *functionName = "getTasksCore";
+
+  debug(DEBUG_FLOW, functionName);
+
+  // Single-core Power PC
+  if (strcmp(cpu_.c_str(), "PowerPC,460EX") == 0) {
+    debug(DEBUG_TRACE, functionName, "CPU type", cpu_.c_str());
+    debug(DEBUG_TRACE, functionName, "Number of cores", cpuNumCores_);
+
+    // CPU Core 0 executes all tasks: Phase, Servo, RT and Background
+    cpuCoreTasks_[PPMAC_CPU_PHASETASK] = 0;
+    cpuCoreTasks_[PPMAC_CPU_SERVOTASK] = 0;
+    cpuCoreTasks_[PPMAC_CPU_RTTASK]    = 0;
+    cpuCoreTasks_[PPMAC_CPU_BGTASK]    = 0;
+
+  // Dual-core Power PC
+  } else if (strcmp(cpu_.c_str(), "PowerPC,APM86xxx") == 0) {
+    debug(DEBUG_TRACE, functionName, "CPU type", cpu_.c_str());
+    debug(DEBUG_TRACE, functionName, "Number of cores", cpuNumCores_);
+
+    // CPU Core 0 executes: Background tasks
+    cpuCoreTasks_[PPMAC_CPU_BGTASK] = 0;
+    // CPU Core 1 executes: Phase, Servo and RT tasks
+    cpuCoreTasks_[PPMAC_CPU_PHASETASK] = 1;
+    cpuCoreTasks_[PPMAC_CPU_SERVOTASK] = 1;
+    cpuCoreTasks_[PPMAC_CPU_RTTASK]    = 1;
+
+  // Dual-core ARM
+  } else if(strcmp(cpu_.c_str(), "arm,LS1021A") == 0) {
+    debug(DEBUG_TRACE, functionName, "CPU type", cpu_.c_str());
+    debug(DEBUG_TRACE, functionName, "Number of cores", cpuNumCores_);
+
+    for (int task_idx = 0; task_idx < 4; task_idx++) {
+      // get task Core command
+      int taskCore;
+      switch (task_idx) {
+        case 0:   strcpy(cmd, "Sys.CorePhase")      ; break;
+        case 1:   strcpy(cmd, "Sys.CoreServo")      ; break;
+        case 2:   strcpy(cmd, "Sys.CoreRti")        ; break;
+        case 3:   strcpy(cmd, "Sys.CoreBackground") ; break;
+      }
+      // get task Core
+      status = pBroker_->immediateWriteRead(cmd, reply);
+      status = parseIntegerVariable(cmd, reply,
+                                    "Read CPU core to execute task", taskCore);
+      if (status == asynSuccess) {
+        cpuCoreTasks_[task_idx] = taskCore;
+      } else {
+        debugf(DEBUG_ERROR, functionName,
```

Comment:
This debug message could state the Sys command which had the issue rather than the task index number